src/main/webapp/definitions/alfresco-model.yaml

```diff
@@ -0,0 +1,475 @@
+security:
+  - basicAuth: []
+consumes:
+  - application/json
+produces:
+  - application/json
+tags:
+  - name: aspects
+    description: Retrieve and manage aspects
+  - name: types
+    description: Retrieve and manage types
+
+paths:
+  '/aspects':
+    get:
+      x-alfresco-since: "7.0.0"
+      tags:
+        - aspects
+      summary: List aspects
+      description: |
+        **Note:** This is available in Alfresco 7.0.0 and newer versions.
+
+        Gets a list of aspects from the data dictionary. The System aspects will be ignored by default.
+        ```JSON
+        {
+          "list": {
+            "pagination": {
+              "count": 0,
+              "hasMoreItems": true,
+              "totalItems": 0,
+              "skipCount": 0,
+              "maxItems": 0
+            },
+            "entries": [
+              {
+                "entry": {
+                  "id": "cm:titled",
+                  "description": "Titled",
+                  "title": "Titled",
+                  "properties": [
+                    {
+                      "id": "cm:title",
+                      "title": "Title",
+                      "description": "Content Title",
+                      "dataType": "d:mltext",
+                      "isMultiValued": false,
+                      "isMandatory": false,
+                      "isMandatoryEnforced": false
+                      "isProtected": false
+                    },
+                    {
+                      ...
+                    }
+                  ]
+                }
+              },
+              {
+                "entry": {
+                  ...
+                }
+              },
+              {
+                "entry": {
+                  ...
+                }
+              },
+            ]
+          }
+        }
+        ```
+      operationId: listAspects
+      parameters:
+        - name: where
+          in: query
+          description: |
+            Optionally filter the list. Here are some examples:
+
+            An aspect should represented in the fully qualified name(`namespace_prefix:name`). e.g 'cm:title'.
+
+            The following where clause will only return aspects from the `namespace1:model` and `namespace2:model`.
+              ```
+              where=(modelIds='namespace1:model,namespace2:model')
+              ```
+
+            The following where clause will only return sub aspects for the given parents.
+              ```
+              where=(parentIds='namespace1:parent,namespace2:parent')
+              ```
+
+            The following where clause will only return aspects that match the pattern.
+              ```
+              where=(uriPrefix matches('http://www.alfresco.org/model.*'))
+              ```
+
+            The following where clause will only return aspects that don't match the pattern.
+              ```
+              where=(not uriPrefix matches('http://www.alfresco.org/model.*'))
+              ```
```

Comment:
We should check that this follows:
- our current REST API guidelines or, if needed, extend those guidelines
- existing examples of "where" clause

In particular:
- using "in" clause for a list of ids ?
- do we have support for "matches" ?
- could/should we use INCLUDESUBTYPES somehow ?
- I see modelId part of the response (I can see parentId in the AbstractClass definition) ?

Reply:
I have created sub tasks for include subtype and show model in the response